csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -332,6 +367,30 @@ private protected Offset<Flatbuf.Schema> SerializeSchema(Schema schema)
                 Builder, endianness, fieldsVectorOffset);
         }
 
+        private protected Offset<Flatbuf.Field>[] GetChildrenFieldOffsets(Field field)
+        {
+            if (!(field.DataType is NestedType type))
+            {
+                return new Offset<Flatbuf.Field>[0];
```

Comment:
Should use `Array.Empty<Offset<Flatbuf.Field>>()` here instead. That will save on allocating a new empty array each time.